### ISSUE TITLE
cloud training disconnects with no useful information as to why

### ISSUE THREAD

**Issue by cherwilco** [ultralytics/hub#1262]
### Search before asking

- [x] I have searched the HUB [issues](https://github.com/ultralytics/hub/issues) and found no similar bug report.

### HUB Component

Datasets

### Bug

https://hub.ultralytics.com/models/E3CdZ5py5t3Q3iklF1X9

trying to train this model from a yolov8 dataset exported from roboflow.

I tried directly uploading the zip that comes from roboflow and the dataset uploads with no errors. I can browse the dataset with no errors... so if that is an invalid dataset or dataset yaml then this is also a bug.... you are accepting the dataset as if its good to go with no warnings or errors. 

when the model disconnects from cloud training there is also no error or warning beyond disconnected! it also will NOT re-start. I would call that buggy behaviour and its been a problem for well over a year... and numerous people have the same problems

**Comment by UltralyticsAssistant** [comment]
👋 Hello @cherwilco, thank you for raising this issue about cloud training disconnects in [Ultralytics HUB](https://hub.ultralytics.com) 🚀  
This is an automated response to help us triage quickly, and an Ultralytics engineer will also assist you soon 👨‍💻🔍

Please visit our [HUB Docs](https://docs.ultralytics.com/hub/) to learn more:
- [**Quickstart**](https://docs.ultralytics.com/hub/quickstart/). Start training and deploying YOLO models with HUB in seconds.
- [**Datasets: Preparing and Uploading**](https://docs.ultralytics.com/hub/datasets/). Learn how to prepare and upload your datasets to HUB in YOLO format.
- [**Projects: Creating and Managing**](https://docs.ultralytics.com/hub/projects/). Group your models into projects for improved organization.
- [**Models: Training and Exporting**](https://docs.ultralytics.com/hub/models/). Train YOLOv5 and YOLOv8 models on your custom datasets and export them to various formats for deployment.
- [**Integrations**](https://docs.ultralytics.com/hub/integrations/). Explore different integration options for your trained models, such as TensorFlow, ONNX, OpenVINO, CoreML, and PaddlePaddle.
- [**Ultralytics HUB App**](https://docs.ultralytics.com/hub/app/). Learn about the Ultralytics App for iOS and Android, which allows you to run models directly on your mobile device.
    * [**iOS**](https://docs.ultralytics.com/hub/app/ios/). Learn about YOLO CoreML models accelerated on Apple's Neural Engine on iPhones and iPads.
    * [**Android**](https://docs.ultralytics.com/hub/app/android/). Explore TFLite acceleration on mobile devices.
- [**Inference API**](https://docs.ultralytics.com/hub/inference-api/). Understand how to use the Inference API for running your trained models in the cloud to generate predictions.

Since this is a 🐛 Bug Report, please share screenshots and clear steps to [reproduce](https://docs.ultralytics.com/help/minimum_reproducible_example/) the disconnect behavior, including a **minimum reproducible example (MRE)** so we can investigate effectively.
If this is a ❓ Question, please provide as much information as possible, including dataset, model, environment details etc. so that we might provide the most helpful response.
We try to respond to all issues as promptly as possible. Thank you for your patience!

**Comment by wifieye-Chad** [comment]
last thing I will provide is the output of the ultralytics dataset verification script I ran locally before uploading. (this is from they ultralytics python package)

PS C:\WINDOWS\system32> python -c "from ultralytics.hub import check_dataset; check_dataset('C:/Users/Chad/Downloads/wifieye V8.v3i.yolov11.zip', task='detect')"
Starting HUB dataset checks for C:\Users\Chad\Downloads\wifieye V8.v3i.yolov11.zip....
WARNING Skipping C:\Users\Chad\Downloads\wifieye V8.v3i.yolov11.zip unzip as destination directory C:\Users\Chad\Downloads\wifieye V8.v3i.yolov11 is not empty.
Fast image access  (ping: 0.00.0 ms, read: 2033.7534.8 MB/s, size: 246.4 KB)
Scanning C:\Users\Chad\Downloads\wifieye V8.v3i.yolov11\detect\labels\train... 2939 images, 0 backgrounds, 0 corrupt: 1
New cache created: C:\Users\Chad\Downloads\wifieye V8.v3i.yolov11\detect\labels\train.cache
Statistics: 100%|██████████| 2939/2939 [00:00<00:00, 833021.99it/s]
Fast image access  (ping: 0.00.0 ms, read: 41.319.6 MB/s, size: 162.9 KB)
Scanning C:\Users\Chad\Downloads\wifieye V8.v3i.yolov11\detect\labels\val... 839 images, 0 backgrounds, 0 corrupt: 100%
New cache created: C:\Users\Chad\Downloads\wifieye V8.v3i.yolov11\detect\labels\val.cache
Statistics: 100%|██████████| 839/839 [00:00<00:00, 837463.36it/s]
Fast image access  (ping: 0.00.0 ms, read: 51.025.1 MB/s, size: 165.0 KB)
Scanning C:\Users\Chad\Downloads\wifieye V8.v3i.yolov11\detect\labels\test... 419 images, 0 backgrounds, 0 corrupt: 100
New cache created: C:\Users\Chad\Downloads\wifieye V8.v3i.yolov11\detect\labels\test.cache
Statistics: 100%|██████████| 419/419 [00:00<00:00, 419530.53it/s]
Checks completed correctly . Upload this dataset to https://hub.ultralytics.com/datasets/.
PS C:\WINDOWS\system32>

**Comment by wifieye-Chad** [comment]
I am actually following your example that is shown upon upload exactly... see screenshot

<img width="789" height="850" alt="Image" src="https://github.com/user-attachments/assets/20878bb8-da41-4502-9400-1566ac9776d5" />

<img width="697" height="330" alt="Image" src="https://github.com/user-attachments/assets/fd288f8d-98d2-4689-a7e2-47bed79dfaaf" />

<img width="644" height="310" alt="Image" src="https://github.com/user-attachments/assets/f62638e4-2051-4335-b732-6573c6b22bac" />

<img width="385" height="301" alt="Image" src="https://github.com/user-attachments/assets/31d99466-134d-4970-89a4-b281bf46a86a" />


Here is my yaml

train: images/train
val: images/val
test: images/test
names:
  0: Car
  1: Person
  2: Tractor
  3: Truck
  4: Van

**Comment by wifieye-Chad** [comment]
can you give me an example?

**Comment by sergiuwaxmann** [comment]
[see example](https://github.com/ultralytics/hub/blob/main/example_datasets/coco8/coco8.yaml#L11)

**Comment by cherwilco** [comment]
even adding path to the yaml exactly as shown I get the same disconnect

https://hub.ultralytics.com/models/4TnczaMhNB1PhbarscYX

there really needs to be some form of logging that the end user can see. whatever is causing this problem is probably some minor thing we could fix on our own.... if we knew what the problem was.... simply showing disconnected is not good.

Imaging trying to offer mechanic advice to someone over the phone when all they will tell you is the car does not work.... nothing more

**Comment by pderrenger** [comment]
@cherwilco (and @wifieye-Chad) — completely agree: a **“Disconnected”** state with no actionable error is not acceptable UX. If the job is failing before the log-stream attaches, the UI can end up showing only “Disconnected”, which leaves you blind. We’re actively pushing to make the **terminal failure reason** visible in the UI.

That said, we can usually narrow this down quickly with two things: **(1) the exact dataset layout that the trainer sees**, and **(2) *any* console tail / timestamp so we can trace the backend job**.

## 1) The key Roboflow gotcha: extra top-level folders (e.g. `detect/`)
Roboflow exports commonly include an extra wrapper directory like `detect/` (your local `check_dataset()` output shows paths like `...\detect\labels\train...`). The Platform upload/browse experience can look fine because it parses/normalizes media, but **training still needs the YAML paths to match the extracted filesystem layout**.

**Recommended for Ultralytics HUB:** zip should have `data.yaml` at the ZIP root and your `images/` + `labels/` at the ZIP root (no extra `detect/` wrapper). See the dataset structure examples in the [Datasets docs](https://docs.ultralytics.com/platform/data/datasets/) and [Platform Quickstart](https://docs.ultralytics.com/platform/quickstart/).

If your YAML is:
```yaml
train: images/train
val: images/val
test: images/test
names:
  0: Car
  1: Person
  2: Tractor
  3: Truck
  4: Van
```
…then the ZIP root must contain:
- `images/train`, `images/val`, `images/test`
- `labels/train`, `labels/val`, `labels/test`
- `data.yaml`

If your ZIP root actually contains `detect/images/train` etc., then either:
- **rezip** so that `images/` and `labels/` are at the root (preferred), **or**
- make YAML consistently point into that folder (but this is exactly where the UI example/screenshot has been confusing folks).

## 2) Local validation is good — but it currently won’t catch this specific HUB mismatch
`from ultralytics.hub import check_dataset(..., task="detect")` is still the right first step and is documented here: see the [`check_dataset()` reference](https://docs.ultralytics.com/reference/hub/__init__/).  
However, as you noticed, it can pass even when the **HUB cloud training environment** will later choke on a layout/YAML mismatch. That’s on us, and we should improve the preflight checks.

Also: your output indicates **1 corrupt item** in the train scan (`0 corrupt: 1`). Even a single bad image/label can crash a dataloader depending on where it hits. Please also check your dataset **Errors** tab in the UI (it explicitly lists processing failures) per the [Datasets docs](https://docs.ultralytics.com/platform/data/datasets/).

## 3) Fastest way to get a real error message today (workaround + diagnosis)
If cloud training won’t surface logs, you can often force a useful stack trace by training against the same uploaded dataset locally using the `ul://` URI (this uses the Platform dataset exactly as stored). Example from the docs: see **Dataset URI** in the [Datasets docs](https://docs.ultralytics.com/platform/data/datasets/).

```bash
yolo train model=yolo26n.pt data=ul://YOUR_USERNAME/datasets/YOUR_DATASET_SLUG epochs=1
```

If this fails, you’ll get an explicit Python error you can paste here (that’s usually enough to pinpoint the cause in minutes).

## 4) What I need from your latest “Disconnected” run to escalate (no guessing)
For this specific model (`https://hub.ultralytics.com/models/4TnczaMhNB1PhbarscYX`), can you reply with:

1) **Exact disconnect timestamp** (include timezone, or UTC).  
2) A screenshot or copy/paste from **Model → Train → Console** (even if it’s only a few lines / blank).  
3) The **dataset URL** used for that run.  
4) The **first ~30 entries of the ZIP file list** (or just confirm whether the ZIP root contains `detect/` or not).

With (1) + (2) we can trace the job runner events; with (3) + (4) we can confirm whether it’s a layout/YAML issue vs an infra issue.

---

If you need a “get it trained right now” path: the Roboflow integration succeeding strongly suggests it’s repackaging the dataset into a layout the trainer expects. Using that integration (or rezipping to remove `detect/` wrappers) is the quickest path while we chase down why the UI is losing the failure reason.
